engine/src/main.cpp

```diff
@@ -716,162 +717,193 @@ void bootstrap_main_loop() {
+        }
+    }
+
+    if (cmd_args.count("mod")) {
+        retstr = cmd_args["mod"].as<std::string>();
+    }
+
+    if (cmd_args.count("start-in-system")) {
+        std::string st;
+        st = cmd_args["start-in-system"].as<std::string>();
+        SetPlayerSystem(st, true);
+    }
+
+    if (cmd_args.count("location")) {
+        QVector PlayerLocation;
+        try {
+            std::string input = cmd_args["location"].as<std::string>();
+            std::istringstream iStringStream(input);
+            iStringStream.ignore(2);
+            iStringStream >> PlayerLocation.i;
+            iStringStream.ignore(1, ',');
+            iStringStream >> PlayerLocation.j;
+            iStringStream.ignore(1, ',');
+            iStringStream >> PlayerLocation.k;
+        } catch (std::ios_base::failure &inputFailure) {
+            std::cout << "Error reading coordinates for player location: " << inputFailure.what()
+                    << std::endl;
+            exit(EXIT_FAILURE);
+        }
+        SetPlayerLoc(PlayerLocation, true);
+    }
+
+    if (cmd_args.count("debug")) {
+        char vs_debug_level_temp = cmd_args["debug"].as<char>();
+        switch (vs_debug_level_temp) {
+            case 0:
+                g_game.vsdebug = 0;
+                break;
+            case 1:
+                g_game.vsdebug = 1;
+                break;
+            case 2:
+                g_game.vsdebug = 2;
+                break;
+            case 3:
+                g_game.vsdebug = 3;
+                break;
+            case '0':
+                g_game.vsdebug = 0;
+                break;
+            case '1':
+                g_game.vsdebug = 1;
+                break;
+            case '2':
+                g_game.vsdebug = 2;
+                break;
+            case '3':
+                g_game.vsdebug = 3;
+                break;
+            default:
+                VS_LOG_FLUSH_EXIT(fatal, "Invalid debug level specified", EXIT_FAILURE);
+        }
+    }
+
+    if (cmd_args.count("net")) {
+        //don't ignore the network section of the config file
+        ignore_network = false;
+    }
+
+    if (cmd_args.count("benchmark")) {
+        benchmark = cmd_args["benchmark"].as<double>();
+    }
+
+    if (cmd_args.count("test-audio")) {
+        // This is handled separately, since we have to undefine main first
+        // return Audio::Test::main(argc, argv);
+    }
+
+    if (cmd_args.count("a")) {
+        g_game.x_resolution = 800;
+        g_game.y_resolution = 600;
+    }
+    if (cmd_args.count("h")) {
+        g_game.x_resolution = 1024;
+        g_game.y_resolution = 768;
+    }
+    if (cmd_args.count("v")) {
+        g_game.x_resolution = 1280;
+        g_game.y_resolution = 1024;
+    }
+
+    if (cmd_args.count("mission_name")) {
+        mission_name = cmd_args["mission_name"].as<std::string>();
+    }
+
     if (false == legacy_data_dir_mode) {
```

Comment:
One day we should figure out this legacy data dir mode. Right now we support this and I'm not sure why.

Reply:
Agreed. Let's discuss with @BenjamenMeyer at some point.